File: leetcode-algorithms/102. Binary Tree Level Order Traversal/solution.py
import queue
def levelOrder(root):
	"""
	:type root: TreeNode
	:rtype: List[List[int]]
	"""
	res = []
	p = root
	q = queue()
	q.put(p)
	while not q.empty():
		for t in q.get():
			if t.left:
				q.put(t.left)
			if t.right:
				q.put(t.right)
			temp.append(t.val)
		res.append(temp)
	return res

File: leetcode-algorithms/102. Binary Tree Level Order Traversal/solution.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> q;
	if(root){
		q.push(root);
	}
	vector<int> inner;
	vector<vector<int> > outer;
	TreeNode *node;
	while(!q.empty()){
		int size = q.size();
		inner.clear();
		while(size-->0){
			node = q.front();
			q.pop();
			inner.push_back(node->val);
			if(node->left)
				q.push(node->left);
			if(node->right)
				q.push(node->right);
		}
		outer.push_back(inner);
	}
	return outer;
    }
};